server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -252,7 +252,6 @@ public IndexScopedSettings copy(Settings settings, IndexMetadata metadata) {
     }
 
     public boolean registerSetting(Setting<?> setting) {
-        validateSettingKey(setting);
         return super.registerSetting(setting);
```

Comment:
Why are we making this change?

Reply:
`super.registerSetting(setting)` calls `validateSettingKey`